### PR TITLE
#3 3-layer architecture

### PR DESCRIPTION
**Pull request by unstoo**
- [x] Install DB PostgreSQL on your machine
- [x] Write SQL script which will create Users table in the DB and fill it in with predefined users’ collection.
- [x] Configure your REST service to work with PostgreSQL
- [x] The service should adhere to 3-layer architecture principles -- `I think I've reached this goal only partially: controllers-validators still look messy/coupled with express framework.`

### CODE REVIEW

File: 2-crud-api/src/data-access/user.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Sequelize } from 'sequelize';
+import dotenv from 'dotenv';
+
+dotenv.config();
```
Comment (danielcsato) [inline 2-crud-api/src/data-access/user.ts:4]: It should be enough to init dotenv only once in your application in `index.ts`. Make sure to put it in line 1

`
import dotenv from 'dotenv';
dotenv.config();

...rest of your index.ts file
`

File: 2-crud-api/src/index.ts [hunk 2]
```diff
@@ -1,78 +1,97 @@
...
-  res.status(StatusCodes.OK).json(id);
-});
-
-app.patch('/patchUser', patchUserSchema, (req: ValidatedRequest<PatchUserSchema>, res: Response) => {
-  const { id, login, password, age } = req.body
-  const user = users[id];
-  if (!user || user.isDeleted) {
-    return res.status(StatusCodes.BAD_REQUEST).json(false);
-  }
+  app.get(Router.getUser.route, getUserSchema, async (req: ValidatedRequest<GetUserSchema>, res: Response) => {
```
Comment (danielcsato) [inline 2-crud-api/src/index.ts:51]: The routes should be under the `controllers` folder , eg.: userRouter.ts , then load the router in the index file.
Example here : https://expressjs.com/en/guide/routing.html     --- express.Router part

**Review by danielcsato — APPROVED** [review]
Left 2 comment, rest looks good 👍

File: 2-crud-api/src/index.ts (hunk 2 above)
Comment (unstoo) [reply 2-crud-api/src/index.ts:52]: re-orginsed code

**Pull request merged**